src/store/RelayRecordWriter.js

```diff
@@ -94,7 +94,7 @@ class RelayRecordWriter {
    */
   getDataID(
     storageKey: string,
-    identifyingArgValue: ?string
+    identifyingArgValue: any
   ): ?DataID {
     if (RelayNodeInterface.isNodeRootCall(storageKey)) {
       invariant(
```

Comment:
If the root call is `node`/`nodes`, then the storage key *must* be a string or int per the Relay GraphQL specification. We can add an invariant here for this and coerce the return value to a string.